src/systems/systems/src/pressurization/cabin_pressure_controller.rs

```diff
@@ -0,0 +1,1304 @@
+
+    fn with_open_outflow_valve() -> Self {
+        Self {
+            vertical_speed: Velocity::new::<foot_per_minute>(0.),
+            timer: Duration::from_secs(Self::OUTFLOW_VALVE_OPENS_AFTER_SECS),
+            pressure_schedule: Ground {
+                cpc_switch_reset: false,
+            },
+        }
+    }
+
+    fn step(
+        self: PressureSchedule<Ground>,
+        context: &UpdateContext,
+        engines: [&impl EngineCorrectedN1; 2],
+    ) -> PressureScheduleManager {
+        if engines
+            .iter()
+            .all(|&x| x.corrected_n1() > Ratio::new::<percent>(70.))
+            && context.is_on_ground()
+        {
+            PressureScheduleManager::TakeOff(self.into())
+        } else if !context.is_on_ground()
+            && context.indicated_airspeed().get::<knot>() > 100.
+            && context.ambient_pressure().get::<hectopascal>() > 0.
+        {
+            PressureScheduleManager::ClimbInternal(self.into())
+        } else {
+            PressureScheduleManager::Ground(self.increase_timer(context))
+        }
+    }
+
+    fn should_open_outflow_valve(&self) -> bool {
+        self.timer >= Duration::from_secs(Self::OUTFLOW_VALVE_OPENS_AFTER_SECS)
+    }
+
+    fn should_switch_cpc(self: PressureSchedule<Ground>) -> bool {
+        self.timer > Duration::from_secs(70) && self.pressure_schedule.cpc_switch_reset
+    }
+
+    fn reset_cpc_switch(&mut self) {
+        self.pressure_schedule.cpc_switch_reset = false;
+    }
+}
+
+transition_with_ctor!(TakeOff, Ground, Ground::reset);
+transition_with_ctor!(DescentInternal, Ground, Ground::reset);
+transition_with_ctor!(Abort, Ground, Ground::reset);
+
+#[derive(Copy, Clone)]
+struct TakeOff;
+
+impl PressureSchedule<TakeOff> {
+    fn step(
+        self: PressureSchedule<TakeOff>,
+        context: &UpdateContext,
+        engines: [&impl EngineCorrectedN1; 2],
+    ) -> PressureScheduleManager {
+        if engines
+            .iter()
+            .all(|&x| x.corrected_n1() < Ratio::new::<percent>(70.))
+            && context.is_on_ground()
+        {
+            PressureScheduleManager::Ground(self.into())
+        } else if !context.is_on_ground()
+            && context.indicated_airspeed().get::<knot>() > 100.
+            && context.ambient_pressure().get::<hectopascal>() > 0.
+        {
+            PressureScheduleManager::ClimbInternal(self.into())
+        } else {
+            PressureScheduleManager::TakeOff(self)
+        }
+    }
+}
+
+transition!(Ground, TakeOff);
+
+#[derive(Copy, Clone)]
+struct ClimbInternal;
+
+impl PressureSchedule<ClimbInternal> {
+    fn step(
+        self: PressureSchedule<ClimbInternal>,
+        context: &UpdateContext,
+    ) -> PressureScheduleManager {
+        if context.indicated_altitude() < Length::new::<foot>(8000.)
+            && context.vertical_speed() < Velocity::new::<foot_per_minute>(-200.)
+        {
+            if self.timer > Duration::from_secs(30) {
```

Comment:
**thought**: It is unclear what is the meaning of `30` and other numbers passed to `Duration::from_secs` within this file.
**suggestion**: Introduce constants. 

**suggestion**: The tests can also use the constants introduced. For example, `31` in the test below can be replaced by the constant + 1, or even better: `Duration::from_secs(Type::CONSTANT_HERE) + Duration::from_millis(1)`.

**suggestion**: Note that `test_bed.run()` defaults to one second. If you want a run without any time, use `test_bed.run_without_delta()`.

```rust
#[test]
    fn cabin_vs_changes_to_cruise() {
        let mut test_bed = SimulationTestBed::new(|_| TestAircraft::new());

        test_bed.run();

        assert!(test_bed.query(|a| a.is_climb()));

        test_bed.set_indicated_altitude(Length::new::<foot>(7900.));
        test_bed.set_vertical_speed(Velocity::new::<foot_per_minute>(99.));
        test_bed.run_with_delta(Duration::from_secs(31));
        test_bed.run();

        assert!(test_bed.query(|a| a.is_cruise()));
        assert_eq!(
            test_bed.query(|a| a.cpc.cabin_vs()),
            Velocity::new::<foot_per_minute>(0.)
        );
    }
```

Reply:
I've added a bunch of constants to clarify what a lot of these numbers mean. The problem is the constants need to be local to each function, as the values change in each one. 

As discussed the tests will need a bit of a refactor, but I'll address that in the next PR where I'll remove a lot of the code duplication and add more clarity in terms of numbers.

Reply:
It is probably possible to make the constants local to each impl block, which would then make it usable in tests e.g.:
```rust
impl PressureSchedule<Cruise> {
    const DURATION_UNTIL_CLIMB: u64 = 60;
}
```

But as you said, let's focus on that on the next PR! 😀